docs/upgrade.html

```diff
@@ -19,25 +19,26 @@
 
 <h3><a id="upgrade" href="#upgrade">1.5 Upgrading From Previous Versions</a></h3>
 
-<h4><a id="upgrade_10_1" href="#upgrade_10_1">Upgrading from 0.10.0.X to 0.10.1.0</a></h4>
+<h4><a id="upgrade_10_1" href="#upgrade_10_1">Upgrading from 0.8.x, 0.9.x or 0.10.0.X to 0.10.1.0</a></h4>
 0.10.1.0 has wire protocol changes. By following the recommended rolling upgrade plan below, you guarantee no downtime during the upgrade.
 However, please notice the <a href="#upgrade_10_1_breaking">Potential breaking changes in 0.10.1.0</a> before upgrade.
 <br>
-Note: Because new protocols are introduced, it is important to upgrade your Kafka clusters before upgrading your clients.
+Note: Because new protocols are introduced, it is important to upgrade your Kafka clusters before upgrading your clients (i.e. 0.10.1.x clients
+only support 0.10.1.x or later brokers while 0.10.1.x brokers also support older clients).
 
 <p><b>For a rolling upgrade:</b></p>
 
 <ol>
     <li> Update server.properties file on all brokers and add the following properties:
         <ul>
-            <li>inter.broker.protocol.version=CURRENT_KAFKA_VERSION (e.g. 0.8.2, 0.9.0.0 or 0.10.0.0).</li>
+            <li>inter.broker.protocol.version=CURRENT_KAFKA_VERSION (e.g. 0.8.2.0, 0.9.0.0 or 0.10.0.0).</li>
             <li>log.message.format.version=CURRENT_KAFKA_VERSION  (See <a href="#upgrade_10_performance_impact">potential performance impact following the upgrade</a> for the details on what this configuration does.)
         </ul>
     </li>
     <li> Upgrade the brokers. This can be done a broker at a time by simply bringing it down, updating the code, and restarting it. </li>
     <li> Once the entire cluster is upgraded, bump the protocol version by editing inter.broker.protocol.version and setting it to 0.10.1.0. NOTE: If your previous message format version is before 0.10.0, you shouldn't touch log.message.format.version yet - this parameter should only change once all consumers have been upgraded to 0.10.0.0 or later.</li>
     <li> Restart the brokers one by one for the new protocol version to take effect. </li>
-    <li> Once all consumers have been upgraded to 0.10.0, change log.message.format.version to 0.10.1 on each broker and restart them one by one.
+    <li> Once all consumers have been upgraded to 0.10.0 or later, change log.message.format.version to 0.10.1 on each broker and restart them one by one.
```

Comment:
Thanks for the clarification. I understand now that I was bringing some misconception into this in my original questions.

If you're upgrading from 0.10.0.X to 0.10.1.0, it sounds like this last step is unnecessary and you can change `log.message.format` to 0.10.1.0 at the same time you change inter.broker.protocol.version.

If that's the case, I'd propose the following:

```
     <li> Upgrade the brokers. This can be done a broker at a time by simply bringing it down, updating the code, and restarting it. </li>
      <li> Once the entire cluster is upgraded, bump the protocol version by editing inter.broker.protocol.version and setting it to 0.10.1.0. 
      <li> If your previous message format is 0.10.0, you can also change log.message.format.version to 0.10.1 at this time. NOTE: If your previous message format version is before 0.10.0, you shouldn't touch log.message.format.version yet - this parameter should only change once all consumers have been upgraded to 0.10.0.0 or later.</li>
      <li> Restart the brokers one by one for the new protocol version to take effect. </li>
      <li> Once all consumers have been upgraded to 0.10.0 or later, change log.message.format.version to 0.10.1 on each broker and restart them one by one. NOTE: If you were able to update the log.message.format.version to 0.10.1 in a previous step, then this restart is unnecessary.
 ```